### PR TITLE
[BugFix] Fix compaction jobs getting stuck when abortCompaction RPC fails

### PR DESCRIPTION
**Pull request by sduzh**
Fix compaction jobs getting stuck when LakeService.abortCompaction throwed exception

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionTask.java [hunk 1]
```diff
@@ -116,10 +112,15 @@ public void sendRequest() {
...
     public void abort() {
         if (!isCompleted()) {
-            LOG.info("abort compaction task, txn_id: {}, node: {}", request.txnId, nodeId);
             AbortCompactionRequest abortRequest = new AbortCompactionRequest();
             abortRequest.txnId = request.txnId;
-            Future<AbortCompactionResponse> ignored = rpcChannel.abortCompaction(abortRequest);
+            try {
+                Future<AbortCompactionResponse> ignored = rpcChannel.abortCompaction(abortRequest);
+                LOG.info("aborted compaction task, txn_id: {}, node: {}", request.txnId, nodeId);
+            } catch (Exception e) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionTask.java:120]: this means the caller will not be aware that the `AbortCompactionRequest` is actually sent out or not, is it expected?

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionTask.java|6|6|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionTask.java (hunk 1 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionTask.java:120]: Yes, it's by design.

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**